helper/schema/grpc_provider.go

```diff
@@ -1544,6 +1563,24 @@ func (s *GRPCProviderServer) ApplyResourceChange(ctx context.Context, req *tfpro
 			return resp, nil
 		}
 
+		if !res.ResourceBehavior.AllowNullIdentity {
+			isFullyNull := true
+			for _, v := range newIdentityVal.AsValueMap() {
```

Comment:
❓ - I'm not 100% sure what the value of `newIdentityVal` will be during a destroy operation (which still calls `ApplyResourceChange`). I'm assuming because of the flatmap this will still just be an object, but could potentially have all null values?

Perhaps there is no logic in providers or the SDKs that actually nulls the values during destroy 🤔 -> I believe state will be nulled out before returning from `ApplyResourceChange` , so this must be different

Reply:
If the operation is a destroy `ApplyResourceChange` will return early and we'll never reach this part of the code

https://github.com/hashicorp/terraform-plugin-sdk/blob/b3515012a37a0648c7cfa2ecdfe50d3ead5fd99b/helper/schema/grpc_provider.go#L1508-L1521